README
NAME
    Test::Perl::Critic - Use Perl::Critic in test programs

SYNOPSIS
    Test one file:

      use Test::Perl::Critic;
      use Test::More tests => 1;
      critic_ok($file);

    Or test all files in one or more directories:

      use Test::Perl::Critic;
      all_critic_ok($dir_1, $dir_2, $dir_N );

    Or test all files in a distribution:

      use Test::Perl::Critic;
      all_critic_ok();

    Recommended usage for CPAN distributions:

      use strict;
      use warnings;
      use File::Spec;
      use Test::More;
      use English qw(-no_match_vars);

      if ( not $ENV{TEST_AUTHOR} ) {
          my $msg = 'Author test.  Set $ENV{TEST_AUTHOR} to a true value to run.';
          plan( skip_all => $msg );
      }

      eval { require Test::Perl::Critic; };

      if ( $EVAL_ERROR ) {
         my $msg = 'Test::Perl::Critic required to criticise code';
         plan( skip_all => $msg );
      }

      my $rcfile = File::Spec->catfile( 't', 'perlcriticrc' );
      Test::Perl::Critic->import( -profile => $rcfile );
      all_critic_ok();

DESCRIPTION
    Test::Perl::Critic wraps the Perl::Critic engine in a convenient
    subroutine suitable for test programs written using the Test::More
    framework. This makes it easy to integrate coding-standards enforcement
    into the build process. For ultimate convenience (at the expense of some
    flexibility), see the criticism pragma.

    If you have an large existing code base, you might prefer to use
    Test::Perl::Critic::Progressive.

    If you'd like to try Perl::Critic without installing anything, there is
    a web-service available at <http://perlcritic.com>. The web-service does
    not yet support all the configuration features that are available in the
    native Perl::Critic API, but it should give you a good idea of what it
    does. You can also invoke the perlcritic web-service from the command
    line by doing an HTTP-post, such as one of these:

      $> POST http://perlcritic.com/perl/critic.pl < MyModule.pm
      $> lwp-request -m POST http://perlcritic.com/perl/critic.pl < MyModule.pm
      $> wget -q -O - --post-file=MyModule.pm http://perlcritic.com/perl/critic.pl

    Please note that the perlcritic web-service is still alpha code. The URL
    and interface to the service are subject to change.

SUBROUTINES
    critic_ok( $FILE [, $TEST_NAME ] )
        Okays the test if Perl::Critic does not find any violations in
        $FILE. If it does, the violations will be reported in the test
        diagnostics. The optional second argument is the name of test, which
        defaults to "Perl::Critic test for $FILE".

        If you use this form, you should emit your own Test::More plan
        first.

    all_critic_ok( [ @DIRECTORIES ] )
        Runs "critic_ok()" for all Perl files beneath the given list of
        @DIRECTORIES. If @DIRECTORIES is empty or not given, this function
        tries to find all Perl files in the blib/ directory. If the blib/
        directory does not exist, then it tries the lib/ directory. Returns
        true if all files are okay, or false if any file fails.

        This subroutine emits its own Test::More plan, so you do not need to
        specify an expected number of tests yourself.

    all_code_files ( [@DIRECTORIES] )
        DEPRECATED: Use the "all_perl_files" subroutine that is exported by
        Perl::Critic::Utils instead.

        Returns a list of all the Perl files found beneath each DIRECTORY,
        If @DIRECTORIES is an empty list, defaults to blib/. If blib/ does
        not exist, it tries lib/. Skips any files in CVS or Subversion
        directories.

        A Perl file is:

        *   Any file that ends in .PL, .pl, .pm, or .t

        *   Any file that has a first line with a shebang containing 'perl'

CONFIGURATION
    Perl::Critic is highly configurable. By default, Test::Perl::Critic
    invokes Perl::Critic with its default configuration. But if you have
    developed your code against a custom Perl::Critic configuration, you
    will want to configure Test::Perl::Critic to do the same.

    Any arguments passed through the "use" pragma (or via
    "Test::Perl::Critic->import()" )will be passed into the Perl::Critic
    constructor. So if you have developed your code using a custom
    ~/.perlcriticrc file, you can direct Test::Perl::Critic to use your
    custom file too.

      use Test::Perl::Critic (-profile => 't/perlcriticrc');
      all_critic_ok();

    Now place a copy of your own ~/.perlcriticrc file in the distribution as
    t/perlcriticrc. Then, "critic_ok()" will be run on all Perl files in
    this distribution using this same Perl::Critic configuration. See the
    Perl::Critic documentation for details on the .perlcriticrc file format.

    Any argument that is supported by the Perl::Critic constructor can be
    passed through this interface. For example, you can also set the minimum
    severity level, or include & exclude specific policies like this:

      use Test::Perl::Critic (-severity => 2, -exclude => ['RequireRcsKeywords']);
      all_critic_ok();

    See the Perl::Critic documentation for complete details on its options
    and arguments.

DIAGNOSTIC DETAILS
    By default, Test::Perl::Critic displays basic information about each
    Policy violation in the diagnostic output of the test. You can customize
    the format and content of this information by using the "-verbose"
    option. This behaves exactly like the "-verbose" switch on the
    perlcritic program. For example:

      use Test::Perl::Critic (-verbose => 6);

      #or...

      use Test::Perl::Critic (-verbose => '%f: %m at %l');

    If given a number, Test::Perl::Critic reports violations using one of
    the predefined formats described below. If given a string, it is
    interpreted to be an actual format specification. If the "-verbose"
    option is not specified, it defaults to 3.

        Verbosity     Format Specification
        -----------   -------------------------------------------------------
         1            "%f:%l:%c:%m\n",
         2            "%f: (%l:%c) %m\n",
         3            "%m at %f line %l\n",
         4            "%m at line %l, column %c.  %e.  (Severity: %s)\n",
         5            "%f: %m at line %l, column %c.  %e.  (Severity: %s)\n",
         6            "%m at line %l, near '%r'.  (Severity: %s)\n",
         7            "%f: %m at line %l near '%r'.  (Severity: %s)\n",
         8            "[%p] %m at line %l, column %c.  (Severity: %s)\n",
         9            "[%p] %m at line %l, near '%r'.  (Severity: %s)\n",
        10            "%m at line %l, column %c.\n  %p (Severity: %s)\n%d\n",
        11            "%m at line %l, near '%r'.\n  %p (Severity: %s)\n%d\n"

    Formats are a combination of literal and escape characters similar to
    the way "sprintf" works. See String::Format for a full explanation of
    the formatting capabilities. Valid escape characters are:

        Escape    Meaning
        -------   ----------------------------------------------------------------
        %c        Column number where the violation occurred
        %d        Full diagnostic discussion of the violation (DESCRIPTION in POD)
        %e        Explanation of violation or page numbers in PBP
        %F        Just the name of the logical file where the violation occurred.
        %f        Path to the logical file where the violation occurred.
        %G        Just the name of the physical file where the violation occurred.
        %g        Path to the physical file where the violation occurred.
        %l        Logical line number where the violation occurred
        %L        Physical line number where the violation occurred
        %m        Brief description of the violation
        %P        Full name of the Policy module that created the violation
        %p        Name of the Policy without the Perl::Critic::Policy:: prefix
        %r        The string of source code that caused the violation
        %C        The class of the PPI::Element that caused the violation
        %s        The severity level of the violation

CAVEATS
    Despite the convenience of using a test script to enforce your coding
    standards, there are some inherent risks when distributing those tests
    to others. Since you don't know which version of Perl::Critic the
    end-user has and whether they have installed any additional Policy
    modules, you can't really be sure that your code will pass the
    Test::Perl::Critic tests on another machine.

    For these reasons, we strongly advise you to make your perlcritic tests
    optional, or exclude them from the distribution entirely.

    The recommended usage in the "SYNOPSIS" section illustrates one way to
    make your perlcritic.t test optional. Another option is to put
    perlcritic.t and other author-only tests in a separate directory (xt/
    seems to be common), and then use a custom build action when you want to
    run them. Also, you should not list Test::Perl::Critic as a requirement
    in your build script. These tests are only relevant to the author and
    should not be a prerequisite for end-use.

    See
    <http://www.chrisdolan.net/talk/index.php/2005/11/14/private-regression-
    tests/> for an interesting discussion about Test::Perl::Critic and other
    types of author-only regression tests.

EXPORTS
      critic_ok()
      all_critic_ok()

PERFORMANCE HACKS
    If you want a small performance boost, you can tell PPI to cache results
    from previous parsing runs. Most of the processing time is in
    Perl::Critic, not PPI, so the speedup is not huge (only about 20%).
    Nonetheless, if your distribution is large, it's worth the effort.

    Add a block of code like the following to your test program, probably
    just before the call to "all_critic_ok()". Be sure to adjust the path to
    the temp directory appropriately for your system.

        use File::Spec;
        my $cache_path = File::Spec->catdir(File::Spec->tmpdir,
                                            "test-perl-critic-cache-$ENV{USER}");
        if (!-d $cache_path) {
           mkdir $cache_path, oct 700;
        }
        require PPI::Cache;
        PPI::Cache->import(path => $cache_path);

    We recommend that you do NOT use this technique for tests that will go
    out to end-users. They're probably going to only run the tests once, so
    they will not see the benefit of the caching but will still have files
    stored in their temp directory.

BUGS
    If you find any bugs, please submit them to
    <http://rt.cpan.org/NoAuth/Bugs.html?Dist=Perl-Critic>. Thanks.

SEE ALSO
    Module::Starter::PBP

    Perl::Critic

    Test::More

CREDITS
    Andy Lester, whose Test::Pod module provided most of the code and
    documentation for Test::Perl::Critic. Thanks, Andy.

AUTHOR
    Jeffrey Ryan Thalhammer <jeff@imaginative-software.com>

COPYRIGHT
    Copyright (c) 2005-2009 Imaginative Software Systems. All rights
    reserved.

    This program is free software; you can redistribute it and/or modify it
    under the same terms as Perl itself. The full text of this license can
    be found in the LICENSE file included with this module.



Changes
#######################################################################
#      $URL: http://perlcritic.tigris.org/svn/perlcritic/tags/Test-Perl-Critic-1.02/Changes $
#     $Date: 2009-10-22 16:03:02 -0700 (Thu, 22 Oct 2009) $
#   $Author: thaljef $
# $Revision: 3687 $
########################################################################

[1.02] Released on 2009-10-22

     Documentation changes.  META.yml "enhancements".  Moved author
     tests out of the way of normal testing.

     Now requires Perl-Critic version 1.105 or newer.  Fixed source
     code to be compliant with this version. Reported by Xavier Caron.

[1.01] Released on 2007-01-24

     PRODUCTION RELEASE: You may now consider the public API of
     Test::Perl::Critic as stable.

     Edited documentation.  Includes better example for using
     Test::Perl::Critic in CPAN distributions.

     Added a t/perlcritic.t test to further demonstrate the typical
     usage for CPAN distributions.

     No code changes.

[0.08] Released on 2006-11-05

     All parameters to Test::Perl::Critic can now be controlled through
     the .perlcriticrc file.  See the Perl::Critic POD for more info.

     Now compatible with the perlcritic.t test script that is generate
     by Module::Starter::PBP.  In my humble opinion, the test script
     is flawed, but at least it will work now.

[0.07] Released on 2006-08-20

     All exceptions during testing are now fatal.

     Now supports -verbose option, just like `perlcritic` does.

     Added support for PPI::Cache, which can speedup repeated tests.
     See Test::Perl::Critic documentation for more details.

     Now requires Perl-Critic version 0.19 or higher.

[0.06] Released on 2006-05-14

     No functionality changes or bug fixes, just refactorings.

     The all_code_files() function is now deprecated.  See
     all_perl_files() in Perl::Critic::Utils instead

     Now requires Perl::Critic version 0.15_03 or higher.

[0.05] Released on 2006-03-26

     Modified import() interface to pass all arguments to Perl::Critic.
     This give you a direct hook to configure Perl::Critic and
     should still be backward compatible.

[0.04] Released on 2006-01-15

     Updated to be compatible with Perl-Critic 0.14, but is not
     backward compatible with older versions of Perl-Critic.  If
     you want to use the -severity option with Perl-Critic, you
     need to use Perl-Critic version 0.14 or later.

[0.03] Released on 2005-10-31

     Now catching exceptions from Perl::Critic so the tests fail
     gracefully if PPI has errors.

     Added -format option, which is similar to the -verbose option
     that perlcritic has.  I'm not sure why I didn't make it exactly
     the same.

     Added a few more test cases.  In general, this module doesn't
     have very good test coverage because I haven't really figured 
     out how to test a Test.

[0.02] Released on 2005-09-30

     Some minor tweaks; Edited POD.

     Removed Test::Pod and Test::Pod::Coverage from the prerequisites 
     list.  Those tests only run if the modules are installed.

     Test::Perl::Critic is going to require Perl::Critic version 0.08 
     or later, or it won't pass its own criticism tests.

[0.01] Not released

     "Plan to throw one away" -Frederick Brooks



lib/Test/Perl/Critic.pm
#######################################################################
#      $URL: http://perlcritic.tigris.org/svn/perlcritic/tags/Test-Perl-Critic-0.08/lib/Test/Perl/Critic.pm $
#     $Date: 2006-11-05 18:46:37 -0800 (Sun, 05 Nov 2006) $
#   $Author: thaljef $
# $Revision: 816 $
########################################################################

package Test::Perl::Critic;

use strict;
use warnings;
use Carp qw(croak);
use English qw(-no_match_vars);
use Test::Builder qw();
use Perl::Critic qw();
use Perl::Critic::Violation qw();
use Perl::Critic::Utils;

our $VERSION = 0.08;

#---------------------------------------------------------------------------

my $TEST        = Test::Builder->new();
my %CRITIC_ARGS = ();

#---------------------------------------------------------------------------

sub import {

    my ( $self, %args ) = @_;
    my $caller = caller;

    no strict 'refs';  ## no critic
    *{ $caller . '::critic_ok' }     = \&critic_ok;
    *{ $caller . '::all_critic_ok' } = \&all_critic_ok;

    $TEST->exported_to($caller);

    # -format is supported for backward compatibility
    if( exists $args{-format} ){ $args{-verbose} = $args{-format}; }
    %CRITIC_ARGS = %args;

    return 1;
}

#---------------------------------------------------------------------------

sub critic_ok {

    my ( $file, $test_name ) = @_;
    croak q{no file specified} if not defined $file;
    croak qq{"$file" does not exist} if not -f $file;
    $test_name ||= qq{Test::Perl::Critic for "$file"};

    my $critic = undef;
    my @violations = ();
    my $ok = 0;

    # Run Perl::Critic
    eval {
        # TODO: Should $critic be a global singleton?
        $critic     = Perl::Critic->new( %CRITIC_ARGS );
        @violations = $critic->critique( $file );
        $ok         = not scalar @violations;
    };

    # Evaluate results
    $TEST->ok( $ok, $test_name );


    if ($EVAL_ERROR) {           # Trap exceptions from P::C
        $TEST->diag( "\n" );     # Just to get on a new line.
        $TEST->diag( qq{Perl::Critic had errors in "$file":} );
        $TEST->diag( qq{\t$EVAL_ERROR} );
    }
    elsif ( not $ok ) {          # Report Policy violations
        $TEST->diag( "\n" );     # Just to get on a new line.
        $TEST->diag( qq{Perl::Critic found these violations in "$file":} );

        my $verbose = $critic->config->verbose();
        Perl::Critic::Violation::set_format( $verbose );
        for my $viol (@violations) { $TEST->diag("$viol") }
    }

    return $ok;
}

#---------------------------------------------------------------------------

sub all_critic_ok {

    my @dirs = @_ ? @_ : _starting_points();
    my @files = all_code_files( @dirs );
    $TEST->plan( tests => scalar @files );

    my $okays = grep { critic_ok($_) } @files;
    return $okays == @files;
}

#---------------------------------------------------------------------------

sub all_code_files {
    my @dirs = @_ ? @_ : _starting_points();
    return Perl::Critic::Utils::all_perl_files(@dirs);
}

#---------------------------------------------------------------------------

sub _starting_points {
    return -e 'blib' ? 'blib' : 'lib';
}

#---------------------------------------------------------------------------

1;


__END__

=pod

=for stopwords API

=head1 NAME

Test::Perl::Critic - Use Perl::Critic in test programs

=head1 SYNOPSIS

  use Test::Perl::Critic;

  critic_ok($file);                          #Test one file
  all_critic_ok($dir_1, $dir_2, $dir_N );    #Test all files in several $dirs
  all_critic_ok()                            #Test all files in distro

=head1 DESCRIPTION

Test::Perl::Critic wraps the L<Perl::Critic> engine in a convenient
subroutine suitable for test programs written using the L<Test::More>
framework.  This makes it easy to integrate coding-standards
enforcement into the build process.  For ultimate convenience (at the
expense of some flexibility), see the L<criticism> pragma.

If you'd like to try L<Perl::Critic> without installing anything,
there is a web-service available at L<http://perlcritic.com>.  The
web-service does not yet support all the configuration features that
are available in the native Perl::Critic API, but it should give you a
good idea of what it does.  You can also invoke the perlcritic
web-service from the command line by doing an HTTP-post, such as one
of these:

    $> POST http://perlcritic.com/perl/critic.pl < MyModule.pm
    $> lwp-request -m POST http://perlcritic.com/perl/critic.pl < MyModule.pm
    $> wget -q -O - --post-file=MyModule.pm http://perlcritic.com/perl/critic.pl

Please note that the perlcritic web-service is still alpha code.  The
URL and interface to the service are subject to change.



=head1 SUBROUTINES

=over 8

=item critic_ok( FILE [, TEST_NAME ] )

Okays the test if Perl::Critic does not find any violations in FILE.
If it does, the violations will be reported in the test diagnostics.
The optional second argument is the name of test, which defaults to
"Perl::Critic test for FILE".

=item all_critic_ok( [@DIRECTORIES] )

Runs C<critic_ok()> for all Perl files beneath the given list of
directories.  If given an empty list, the function tries to find all
Perl files in the F<blib/> directory.  If the F<blib/> directory does
not exist, then it tries the F<lib/> directory.  Returns true if all
files are okay, or false if any file fails.

If you are building a module with the usual CPAN directory structure,
just make a F<t/perlcritic.t> file like this:

  use Test::Perl::Critic;
  all_critic_ok();

Or if you use the latest version of L<Module::Starter::PBP>, it will
generate this and several other standard test programs for you.

=item all_code_files ( [@DIRECTORIES] )

B<DEPRECATED:> Use the C<all_perl_files> subroutine that is exported
by L<Perl::Critic::Utils> instead.

Returns a list of all the Perl files found beneath each DIRECTORY, If
@DIRECTORIES is an empty list, defaults to F<blib/>.  If F<blib/> does
not exist, it tries F<lib/>.  Skips any files in CVS or Subversion
directories.

A Perl file is:

=over 4

=item * Any file that ends in F<.PL>, F<.pl>, F<.pm>, or F<.t>

=item * Any file that has a first line with a shebang containing 'perl'

=back

=back

=head1 CONFIGURATION

L<Perl::Critic> is highly configurable.  By default,
Test::Perl::Critic invokes Perl::Critic with it's default
configuration.  But if you have developed your code against a custom
Perl::Critic configuration, you will want to configure
Test::Perl::Critic to do the same.

Any arguments given to the C<use> pragma will be passed into the
L<Perl::Critic> constructor.  So if you have developed your code using
a custom F<~/.perlcriticrc> file, you can ask Test::Perl::Critic to
use a custom file too.

  use Test::Perl::Critic (-profile => 't/perlcriticrc');
  all_critic_ok();

Now place a copy of your own F<~/.perlcriticrc> file in the distribution
as F<t/perlcriticrc>.  Then, C<critic_ok()> will be run on all Perl
files in this distribution using this same Perl::Critic configuration.
See the L<Perl::Critic> documentation for details on the
F<.perlcriticrc> file format.

Any argument that is supported by the L<Perl::Critic> constructor can
be passed through this interface.  For example, you can also set the
minimum severity level, or include & exclude specific policies like
this:

  use Test::Perl::Critic (-severity => 2, -exclude => ['RequireRcsKeywords']);
  all_critic_ok();

See the L<Perl::Critic> documentation for complete details on it's
options and arguments.

=head1 DIAGNOSTIC DETAILS

By default, Test::Perl::Critic displays basic information about each
Policy violation in the diagnostic output of the test.  You can
customize the format and content of this information by giving an
additional C<-verbose> option to the C<use> pragma.  This behaves
exactly like the C<-verbose> switch on the F<perlcritic> program.  For
example:

  use Test::Perl::Critic (-verbose => 6);

  #or...

  use Test::Perl::Critic (-verbose => '%f: %m at %l');

If given a number, Test::Perl::Critic reports violations using one of
the predefined formats described below. If given a string, it is
interpreted to be an actual format specification. If the -verbose
option is not specified, it defaults to 3.

    Verbosity     Format Specification
    -----------   -------------------------------------------------------------
     1            "%f:%l:%c:%m\n",
     2            "%f: (%l:%c) %m\n",
     3            "%m at %f line %l\n",
     4            "%m at line %l, column %c.  %e.  (Severity: %s)\n",
     5            "%f: %m at line %l, column %c.  %e.  (Severity: %s)\n",
     6            "%m at line %l, near '%r'.  (Severity: %s)\n",
     7            "%f: %m at line %l near '%r'.  (Severity: %s)\n",
     8            "[%p] %m at line %l, column %c.  (Severity: %s)\n",
     9            "[%p] %m at line %l, near '%r'.  (Severity: %s)\n",
    10            "%m at line %l, column %c.\n  %p (Severity: %s)\n%d\n",
    11            "%m at line %l, near '%r'.\n  %p (Severity: %s)\n%d\n"

Formats are a combination of literal and escape characters similar to
the way sprintf works. See String::Format for a full explanation of
the formatting capabilities. Valid escape characters are:

    Escape    Meaning
    -------   ----------------------------------------------------------------
    %c        Column number where the violation occurred
    %d        Full diagnostic discussion of the violation
    %e        Explanation of violation or page numbers in PBP
    %f        Name of the file where the violation occurred.
    %l        Line number where the violation occurred
    %m        Brief description of the violation
    %P        Name of the Policy module that created the violation
    %p        Name of the Policy without the Perl::Critic::Policy:: prefix
    %r        The string of source code that caused the violation
    %s        The severity level of the violation

=head1 CAVEATS

Despite the obvious convenience of using test programs to verify that
your code complies with coding standards, it is not really sensible to
distribute your module with those test programs.  You don't know which
version of Perl::Critic the user has and whether they have installed
additional Policy modules, so you can't really be sure that your code
will pass the Test::Perl::Critic tests on another machine.

The easy solution is to add your F<perlcritic.t> test program to the
F<MANIFEST.SKIP> file.  When you test your build, you'll still be able
to run the Perl::Critic tests with C<"make test">, but they won't be
included in the tarball when you C<"make dist">.

See L<http://www.chrisdolan.net/talk/index.php/2005/11/14/private-regression-tests/>
for an interesting discussion about Test::Perl::Critic and other types
of author-only regression tests.

=head1 EXPORTS

  critic_ok()
  all_critic_ok()

=head1 PERFORMANCE HACKS

If you want a small performance boost, you can tell PPI to cache
results from previous parsing runs.  Most of the processing time is in
Perl::Critic, not PPI, so the speedup is not huge (only about 20%).
Nonetheless, if your distribution is large, it's worth the effort.

Add a block of code like the following to your test program, probably
just before the call to C<all_critic_ok()>.  Be sure to adjust the
path to the temp directory appropriately for your system.

    use File::Spec;
    my $cache_path = File::Spec->catdir(File::Spec->tmpdir,
                                        "test-perl-critic-cache-$ENV{USER}");
    if (!-d $cache_path) {
       mkdir $cache_path, oct 700;
    }
    require PPI::Cache;
    PPI::Cache->import(path => $cache_path);

We recommend that you do NOT use this technique for tests that will go
out to end-users.  They're probably going to only run the tests once,
so they will not see the benefit of the caching but will still have
files stored in their temp directory.

=head1 BUGS

Please report all bugs to L<http://rt.cpan.org>.  Thanks.

=head1 SEE ALSO

L<Module::Starter::PBP>

L<Perl::Critic>

L<Test::More>

=head1 CREDITS

Andy Lester, whose L<Test::Pod> module provided most of the code and
documentation for Test::Perl::Critic.  Thanks, Andy.

=head1 AUTHOR

Jeffrey Ryan Thalhammer <thaljef@cpan.org>

=head1 COPYRIGHT

Copyright (c) 2005-2006 Jeffrey Ryan Thalhammer.  All rights reserved.

This program is free software; you can redistribute it and/or modify
it under the same terms as Perl itself.  The full text of this license
can be found in the LICENSE file included with this module.

=cut


Makefile.PL
#######################################################################
#      $URL: http://perlcritic.tigris.org/svn/perlcritic/tags/Test-Perl-Critic-1.02/Makefile.PL $
#     $Date: 2008-06-17 12:27:21 -0700 (Tue, 17 Jun 2008) $
#   $Author: clonezone $
# $Revision: 2450 $
########################################################################

use 5.006001;

use strict;
use warnings;

use ExtUtils::MakeMaker;

WriteMakefile(
    NAME           => 'Test::Perl::Critic',
    AUTHOR         => 'Jeffrey Thalhammer <thaljef@cpan.org>',
    ABSTRACT_FROM  => 'lib/Test/Perl/Critic.pm',
    VERSION_FROM   => 'lib/Test/Perl/Critic.pm',
    PL_FILES       => {},
    PREREQ_PM      => {
        'Carp'                      => 0,
        'English'                   => 0,
        'Perl::Critic'              => 1.082,
        'Perl::Critic::Utils'       => 1.082,
        'Perl::Critic::Violation'   => 1.082,
        'strict'                    => 0,
        'Test::Builder'             => 0,
        'warnings'                  => 0,

        'Test::More'              => 0,
    },
);


Build.PL
#######################################################################
#      $URL: http://perlcritic.tigris.org/svn/perlcritic/tags/Test-Perl-Critic-1.02/Build.PL $
#     $Date: 2009-10-22 16:25:08 -0700 (Thu, 22 Oct 2009) $
#   $Author: thaljef $
# $Revision: 3689 $
########################################################################

use 5.006001;

use strict;
use warnings;

use Module::Build;


my $class = Module::Build->subclass( code => <<'END_SUBCLASS' );

    sub ACTION_test {
        my ($self) = @_;

        $self->depends_on('manifest');

        return $self->SUPER::ACTION_test();
    }

    sub ACTION_authortest {
        my ($self) = @_;

        $self->depends_on('build');
        $self->depends_on('manifest');
        $self->depends_on('distmeta');

        $self->test_files( qw< t xt/author > );
        $self->recursive_test_files(1);

        $self->depends_on('test');

        return;
    } # end ACTION_authortest()

    sub ACTION_distdir {
        my ($self) = @_;

        $self->depends_on('authortest');

        return $self->SUPER::ACTION_distdir;
    } # end ACTION_distdir

END_SUBCLASS

my $builder = $class->new(
    module_name     => 'Test::Perl::Critic',
    dist_author     => 'Jeffrey Thalhammer <thaljef@cpan.org>',
    dist_abstract   => 'Use Perl::Critic in test programs.',
    license         => 'perl',
    dynamic_config  => 1,
    create_readme   => 1,
    create_packlist => 1,
    sign            => 0,

    requires        => {
        'Carp'                      => 0,
        'English'                   => 0,
        'Perl::Critic'              => 1.105,
        'Perl::Critic::Utils'       => 1.105,
        'Perl::Critic::Violation'   => 1.105,
        'strict'                    => 0,
        'Test::Builder'             => 0,
        'warnings'                  => 0,
    },

    build_requires       => {
        'Test::More'                => 0,
    },

    meta_merge      => {
        resources   => {
            homepage    => 'http://perlcritic.com',
            license     => 'http://dev.perl.org/licenses/',
            Repository  => 'http://guest@perlcritic.tigris.org/svn/perlcritic/trunk/distributions/Test-Perl-Critic/',
            MailingList => 'http://perlcritic.tigris.org/servlets/SummarizeList?listName=users',
        }
    },
);

$builder->create_build_script();
